crates/egui/src/widgets/slider.rs

```diff
@@ -720,12 +744,16 @@ impl<'a> Slider<'a> {
         }
     }
 
-    fn handle_radius(&self, rect: &Rect) -> f32 {
+    fn handle_radius(&self, rect: &Rect, handle_shape: &style::HandleShape) -> f32 {
         let limit = match self.orientation {
             SliderOrientation::Horizontal => rect.height(),
             SliderOrientation::Vertical => rect.width(),
         };
-        limit / 2.5
+        let aspect_ratio = match handle_shape {
+            style::HandleShape::Circle => 1.0,
+            style::HandleShape::Rect { aspect_ratio } => *aspect_ratio,
+        };
+        limit / (2.5 / aspect_ratio)
```

Comment:
If we define the radius to be on the cross axis (i.e. the Y axis on a horizontal slider), then we don't need to use `aspect_ratio` here